platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -17,68 +19,374 @@ components within the platform.
+5. `memo` - used to store the account number in the Hedera Address Book and is not relevant to the platform
+6. `internal ip address and port` - This will eventually be replaced with node specific configuration.
+
+In order to make a smooth transition from the use of config.txt to using a roster stored in the state, the fields
+related to `internal` ip address and port are being stored in the list of gossip network endpoints in the roster. A
+platform design proposal in the near future will move the `internal` ip address and port field to node configuration.
+In the end, there will only be external endpoints in the roster. In this proposal, we only connect to a node's
+external gossip endpoint. The internal endpoint is ignored.
 
-## What this proposal does not address
+The use of config.txt for starting new networks from genesis and transplanting merkle state to a new network will
+be replaced and require the presence of specific files on disk to initiate these scenarios. A `genesis-config.txt`
+file will be used to start a new network from genesis, and an `override-config.txt` file will be used to transplant
+merkle state to a new network. Initiating these scenarios through distinct files will make the logic for
+determining which roster to use at startup more straightforward.
 
-## Requirements
+To be consistent with future consensus design, the running platform code will not directly read from or write to the
+merkle state with respect to roster data. The appropriate roster data will be provided to the platform during platform
+construction.
 
-- Immutability: The roster hash will be a critical piece of data. Rosters must be immutable to protect the integrity
-  of its computed hash.
-- Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and unnecessary data
-  copying.
-- Clear API: A well-defined API should be provided for the App to create and submit candidate rosters to the platform,
-  and for the platform to manage the lifecycle of submitted rosters.
-- Components update: Platform components that currently rely on the use of the Address book (such as Reconnect, Network,
-  Event Validation etc.) must be adapted to use the new `Roster` object instead.
-- Address book paradigm discontinued within the platform codebase in favor of `Roster`s
+#### Alternatives Considered
 
-## Architecture
+Continuing to use config.txt as the medium for updating the platform `AddressBook` was considered. However, the
+discrepancy in the information present between the `Roster` data structure and the `AddressBook` data structure make
+config.txt a poor medium for carrying out `Roster` updates. Writing the updated roster out to disk is less secure
+than storing it in the state. When TSS is implemented, there will be TSS related key material stored in the state,
+and it is imperative that this key material stay in sync with the `Roster` data. Leaving the content as a modifiable
```

Comment:
This text is already exists verbatim in the Purpose and Context section. Consider deduplicating or re-wording?